src/lex.rs

```diff
@@ -122,7 +129,11 @@ lazy_static! {
 }
 
 fn factorial(n: f64) -> f64 {
-    n.signum() * (1..=n.abs() as u64).fold(1, |p, n| p * n) as f64
+    let answer = (1..=n.round() as u128).map(|u| u as f64).product::<f64>() as f64;
+    if answer == 0.0 {
+        return f64::INFINITY;
+    }
```

Comment:
Why is this infinity if it's zero? So 0.0 is not a valid value?

Reply:
I don't think 0.0 is in the domain of the discrete factorial function. It'd be nice to add support for the gamma function but that's not in the scope of this PR.

Reply:
However, you did make me notice that I didn't handle 0 as an input value, I think I'll just hardcode that as 1.